src/Octoshift/AwsApi.cs

```diff
@@ -0,0 +1,66 @@
+﻿using System;
+using System.IO;
+using System.Threading.Tasks;
+using Amazon;
+using Amazon.S3;
+using Amazon.S3.Model;
+using Amazon.S3.Transfer;
+
+namespace OctoshiftCLI;
+
+public class AwsApi : IDisposable
+{
+    private const int AUTHORIZATION_TIMEOUT_IN_HOURS = 24;
+    private static readonly RegionEndpoint RegionEndpoint = RegionEndpoint.USEast1;
+
+    private readonly ITransferUtility _transferUtility;
+
+#pragma warning disable CA2000
+    public AwsApi(string awsAccessKey, string awsSecretKey) : this(new TransferUtility(new AmazonS3Client(awsAccessKey, awsSecretKey, RegionEndpoint)))
+#pragma warning restore CA2000
+    {
+    }
+
+    internal AwsApi(ITransferUtility transferUtility) => _transferUtility = transferUtility;
+
+    public virtual async Task<string> UploadFileToBucket(string bucketName, string fileName, string keyName)
```

Comment:
Not a huge concern and blocker but we can use method overloading here and keep the method names the same, exactly how `TransferUtility#UploadAsync()` is overloaded. So this way we will have the following methods:
```c#
public virtual async Task<string> UploadToBucket(string bucketName, string fileName, string keyName)
{
    ...
}

public virtual async Task<string> UploadToBucket(string bucketName, byte[] bytes, string keyName)
{
    ...
}
```

I will leave the decision to change up to you though.